phoenix-core/src/it/java/org/apache/phoenix/end2end/json/JsonFunctionsIT.java

```diff
@@ -749,4 +761,186 @@ public void testJsonWithSetGetObjectAPI() throws Exception {
             assertFalse(rs.next());
         }
     }
+
+    @Test
+    public void testSimpleJsonModifyWithAutoCommit() throws Exception {
+        Properties props = PropertiesUtil.deepCopy(TEST_PROPERTIES);
+        String tableName = generateUniqueName();
+        try (Connection conn = DriverManager.getConnection(getUrl(), props)) {
+            conn.setAutoCommit(true);
+            String ddl = "create table " + tableName +
+                    " (pk integer primary key, " +
+                    "col integer, " +
+                    "strcol varchar, " +
+                    "strcol1 varchar, " +
+                    "strcol2 varchar, " +
+                    "strcol3 varchar, " +
+                    "strcol4 varchar, " +
+                    "strcol5 varchar, " +
+                    "jsoncol json)";
+            conn.createStatement().execute(ddl);
+            PreparedStatement stmt = conn.prepareStatement("UPSERT INTO " + tableName + " (pk,col,strcol,jsoncol) VALUES (?,?,?,?)");
+            stmt.setInt(1, 1);
+            stmt.setInt(2, 2);
+            stmt.setString(3, "");
+            stmt.setString(4, basicJson);
+            stmt.execute();
+            String upsert = "UPSERT INTO " + tableName + "(pk,col,strcol,jsoncol) VALUES(1,2,JSON_VALUE(jsoncol, '$.info.address.town'),JSON_MODIFY(jsoncol, '$.info.address.town', '\"Manchester\"')) ";
```

Comment:
So this query doesn't work:
`UPSERT INTO TABLENAME (pk,col,strcol,jsoncol) VALUES(1,2,JSON_VALUE(jsoncol, '$.info.address.town') || 'City',JSON_MODIFY(jsoncol, '$.info.address.town', 'Manchester')`

Reply:
But if I change the query to `UPSERT INTO TABLENAME (pk) VALUES (1) ON DUPLICATE KEY UPDATE strcol = JSON_VALUE(jsoncol, '$.info.address.town') || 'City', jsoncol = JSON_MODIFY(jsoncol, '$.info.address.town', 'Manchester')` then it works

Reply:
what does this mean? "|| 'City'"?

Reply:
I will add this case

Reply:
I will create a separate JIRA to handle nested JSON Functions.

Reply:
I have handled this scenario